### PR TITLE
Fix EC2 filter implementations

### PR DESCRIPTION
**Pull request by kyleaa**
As discussed in https://github.com/CargoSense/ex_aws/issues/241, the filter implementations on the describe methods do not work correctly.

Since I need `describe_tags` I started with that one, and if this looks good I can implement changes in other methods. I borrowed inspiration from JS SDK 
http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EC2.html#describeTags-property

### CODE REVIEW

File: test/lib/ex_aws/ec2/integration_test.exs [hunk 1]
```diff
@@ -5,4 +5,21 @@ defmodule ExAws.EC2.IntegrationTest do
     assert {:ok, %{body: body}} = ExAws.EC2.describe_instances |> ExAws.request
     assert body |> String.contains?("DescribeInstancesResponse")
   end
+
+  test "#describe_tags" do
```
Comment (kyleaa) [inline test/lib/ex_aws/ec2/integration_test.exs]: I used these for my own sanity testing, but they could potentially leak data (if the request succeeds but the pattern match fails) through travis CI if they were given describe on a * resource - I'm not sure if those tests are isolated in their own AWS account.

**Comment by kyleaa** [comment]
I moved the filter building into `normalize_opts` since it then fixes all of the filtered endpoints, which are numerous. How detailed do we want the typespecs to be? They currently are incredibly detailed (but don't work), so I was just curious before diving into fixing those.

I also presume all of the other `_1` typespec'd parameters need similar treatment - take a list, automatically apply the `.N` counts.

File: test/lib/ex_aws/ec2/integration_test.exs (hunk 1 above)
Comment (benwilson512) [inline test/lib/ex_aws/ec2/integration_test.exs]: Is it necessary to do actual requests here? Can't you just assert on the request body?

File: test/lib/ex_aws/ec2_test.exs [hunk 2]
```diff
@@ -1609,6 +1609,41 @@ test "delete_key_pair with options" do
...
+          "Filter.1.Name" => "resource-type",
+          "Filter.1.Value.1" => "customer-gateway",
+          "Filter.2.Name" => "resource-type",
+          "Filter.2.Value.1" => "customer-gateway",
+          "Filter.2.Value.2" => "instance",
+        },
+        path: "/",
+        http_method: :get
+      }
+      assert expected == EC2.describe_tags filters: [{:resource_type, [:customer_gateway]},{:resource_type, [:customer_gateway, :instance]}]
```
Comment (benwilson512) [inline test/lib/ex_aws/ec2_test.exs]: normal keyword syntax should work fine here right? `[resource_type: [:customer_gateway], etc`

Comment (kyleaa) [inline test/lib/ex_aws/ec2_test.exs]: Yes it does, I'll update the tests to reflect.

Comment (kyleaa) [inline test/lib/ex_aws/ec2_test.exs]: re: integration tests, yeah they're probably unnecessary - I used them to sanity-check that it was working.

I'll go through and clean up the typespecs and other endpoints.

**Comment by benwilson512** [comment]
Thanks!

**Pull request merged**